docs/api-spec-swagger.yaml

```diff
@@ -0,0 +1,3370 @@
+      tags:
+        - job
+      summary: Submit a job for admin approval
+      description: Allows users to submit a job listing for admin approval. 
+      operationId: submitJobForApproval
+      security:
+        - jwt: []
+      requestBody:
+        description: Job details to be submitted for approval
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                title:
+                  type: string
+                  example: Frontend Developer
+                level_id:
+                  type: string
+                  example: 147cfbf5-ab6f-4530-9293-213a881a31fd
+                category_id :
+                  type: string
+                  example: 147cfbf5-ab6f-4530-9293-213a881a31fd
+                type_id:
+                  type: string
+                  example: 147cfbf5-ab6f-4530-9293-213a881a31fd
+                location:
+                  type: string
+                  example: Beirut
+                description:
+                  type: string
+                  example: Frontend position 
+                compensation:
+                  type: string
+                  example: 1200$-1500$
+                application_link:
+                  type: string
+                  example: https://www.google.com
+                is_external:
+                  type: boolean
+                  example: true
+                tags:
+                  type: string
+                  example: '#frontend,#react'
+                
+      responses:
+        '202':
+          description: Job submitted for approval
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Job submitted for approval
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Invalid input
+                  
+        
+        '401':
+          description: Unauthorized - Invalid token
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Unauthorized - Invalid token
+                  
+        '403':
+          description: Forbidden - User does not have permission
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: Forbidden - User does not have permission
+                  
+  
+  /jobs/{jobId}{publish}/approve:

```

Comment:
can we remove publish here?

Reply:
Should it be moved to the request body on the Job post endpoint? It indicates whether the job post should be sent to Whatsapp or not.